### PR TITLE
feat: adding svelte component testing support

### PR DESCRIPTION
**Pull request by ZachJW34**
### User facing changelog
Svelte Component Testing Support

### Additional details
Merging of the Svelte Feature branch into develop, no additional changes from the two previously merged and approved PRS: https://github.com/cypress-io/cypress/pull/23519 and https://github.com/cypress-io/cypress/pull/23441

Documentation is WIP: https://github.com/cypress-io/cypress-documentation/issues/4677

### Steps to test
You can scaffold a new svelte project with `npm init vite` and choose Svelte (the scaffold has outdated peer deps so update `@sveltejs/vite-plugin-svelte` and `vite` to latest). Then you can point `yarn cypress:open` at the svelte project and get to testing!

There are also some system tests in `system-tests/test/component_testing_spec.ts` that test both webpack and vite.

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/186748441-30b696c1-1bda-429c-aaa8-de2d439508ac.mov

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by ZachJW34** [comment]
I fixed up the types in [e53f3a1](https://github.com/cypress-io/cypress/pull/23553/commits/e53f3a11616da5698bf9a5bebb6666daee423cdb)

Before:
Screen Shot 2022-08-25 at 8 57 03 PM

After:
Screen Shot 2022-08-25 at 8 57 42 PM

**Review by lmiller1990 — APPROVED** [review]
I approved all previous PRs, this looks to be 💯 

Couple of failing CI steps, but looks like Percy just failing to finalize.

File: npm/svelte/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# @cypress/svelte
+
+Mount Svelte components in the open source [Cypress.io](https://www.cypress.io/) test runner **v10.6.0+**
```
Comment (mike-plummer) [inline npm/svelte/README.md:3]: Should be 10.7.0?

File: npm/svelte/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+> **Note:** This package is bundled with the `cypress` package and should not need to be installed separately. See the [Svelte Component Testing Docs](https://docs.cypress.io/guides/component-testing/quickstart-svelte#Configuring-Component-Testing) for mounting Svelte components. Installing and importing `mount` from `@cypress/svelte` should only be used for advanced use-cases.
+
+## Install
+
+- Requires Svelte >= 3
+- Requires Cypress v10.6.0 or later
```
Comment (mike-plummer) [inline npm/svelte/README.md:10]: version number incorrect

File: npm/svelte/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+npx cypress run --component
+```
+
+For more information, please check the official docs for [running Cypress](https://on.cypress.io/guides/getting-started/opening-the-app#Quick-Configuration) and for [component testing](https://on.cypress.io/guides/component-testing/writing-your-first-component-test).
+
+## Example
+
+```js
+import { mount } from '@cypress/svelte'
+import { HelloWorld } from './HelloWorld.svelte'
```
Comment (mike-plummer) [inline npm/svelte/README.md:35]: Should be the default export, right?

File: system-tests/project-fixtures/svelte/cypress/support/component.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// You can change the location of this file or turn off
+// automatically serving support files with the
+// 'supportFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import "./commands";
```
Comment (mike-plummer) [inline system-tests/project-fixtures/svelte/cypress/support/component.js:17]: /nitpick: `commands` file looks to be empty, could we just remove it and this import?

**Review by mike-plummer — APPROVED** [review]
Amazing work! 💯 Validated Vite JS/TS and Webpack, all bootstrap and run tests as expected. Just a couple minor issues on the docs

File: npm/svelte/README.md (hunk 1 above)
Comment (ZachJW34) [reply npm/svelte/README.md]: Addressed in [b8cd5fe](https://github.com/cypress-io/cypress/pull/23553/commits/b8cd5fe1702b153d9d59a43775dffa63a5442178)

File: npm/svelte/README.md (hunk 3 above)
Comment (ZachJW34) [reply npm/svelte/README.md]: Addressed in [b8cd5fe](https://github.com/cypress-io/cypress/pull/23553/commits/b8cd5fe1702b153d9d59a43775dffa63a5442178)

File: system-tests/project-fixtures/svelte/cypress/support/component.js (hunk 4 above)
Comment (ZachJW34) [reply system-tests/project-fixtures/svelte/cypress/support/component.js]: Addressed in [b8cd5fe](https://github.com/cypress-io/cypress/pull/23553/commits/b8cd5fe1702b153d9d59a43775dffa63a5442178)

**Pull request merged**